src/cmdtab.c

```diff
@@ -1441,14 +1451,13 @@ static void OnWindowPaint(void)
 	// BeginPaint passed our 'hwnd')
 	PAINTSTRUCT ps = {0};
 	handle context = BeginPaint(Switcher, &ps);
-	if (ps.fErase) {
-	}
+	EnterCriticalSection(&lock);
 	BitBlt(context, 0, 0, DrawingRect.right - DrawingRect.left, DrawingRect.bottom - DrawingRect.top, DrawingContext, 0, 0, SRCCOPY);
+	LeaveCriticalSection(&lock);
 	//int scale = 1.0;
 	//SetStretchBltMode(DrawingContext, HALFTONE);
 	//StretchBlt(hdc, 0, 0, gdi_width, gdi_height, DrawingContext, 0, 0, gdi_width * scale, gdi_height * scale, SRCCOPY);
 	EndPaint(Switcher, &ps);
-	ReleaseDC(Switcher, context);
```

Comment:
EndPaint() did release context already. Not sure if this has been the reason for crashes.